### ISSUE TITLE
`grails-gsp` should not depend on `codeconsole/grails-sitemesh3`

### ISSUE THREAD

**Issue by matrei** [grails/grails-gsp#465]
The dependency appears to be required solely for the `String` constant `RENDERING_VIEW_ATTRIBUTE`.

@codeconsole Could we either reverse this dependency or recreate the constant in `grails-gsp` to decouple `grails-gsp` from the `grails-sitemesh3` plugin?

**Comment by codeconsole** [comment]
@matrei I believe this was done because there was a forced dependency for the constant for SiteMesh 2.  Can you provide a code link to where the plugin is being required?

**Comment by matrei** [comment]
Sure, here are links to the usages:

https://github.com/grails/grails-gsp/blob/acfd8d699b7cc4fe78e482245cd052a6ab66bcf7/grails-web-gsp/build.gradle#L7

https://github.com/grails/grails-gsp/blob/acfd8d699b7cc4fe78e482245cd052a6ab66bcf7/grails-web-gsp/src/main/groovy/org/grails/web/pages/GroovyPagesServlet.java#L22

https://github.com/grails/grails-gsp/blob/acfd8d699b7cc4fe78e482245cd052a6ab66bcf7/grails-web-gsp/src/main/groovy/org/grails/web/pages/GroovyPagesServlet.java#L219

https://github.com/grails/grails-gsp/blob/acfd8d699b7cc4fe78e482245cd052a6ab66bcf7/grails-web-gsp/src/main/groovy/org/grails/web/servlet/view/GroovyPageView.java#L37

https://github.com/grails/grails-gsp/blob/acfd8d699b7cc4fe78e482245cd052a6ab66bcf7/grails-web-gsp/src/main/groovy/org/grails/web/servlet/view/GroovyPageView.java#L70

**Comment by codeconsole** [comment]
@matrei thanks for that. It has been so long, I forgot I removed sitemesh2 from grails-gsp as well.  The reason it was required is layouts are a fundamental part of Grails Web GSP and they will not work without SiteMesh or SiteMesh 3.  SiteMesh 2 has always been a forced dependency.  If we want to offer a version of grails that does not support layouts, we could remove the dependency. I don't have a problem with that.  Another reason to not force SiteMesh 3 resolution would be if we decided to keep support for SiteMesh 2.

**Comment by codeconsole** [comment]
sitemesh3-grails-plugin could also be brought in under grails-gsp repo

**Comment by matrei** [comment]
> sitemesh3-grails-plugin could also be brought in under grails-gsp repo

@codeconsole Yes, I think that makes sense, as sitemesh is such an integral part of Grails 👍

**Comment by codeconsole** [comment]
https://github.com/grails/grails-gsp/commit/cf0265c1dd853cab131b58ec81e99fe2462dca25

**Comment by jdaugherty** [comment]
@codeconsole @jamesfredley if we bring the sitemesh plugin under grails gsp, we remove one of the last 2 circular dependency references.  We should consider doing this to stream line our build process

**Comment by jamesfredley** [comment]
https://github.com/grails/grails-gsp/tree/7.0.x/grails-plugin-sitemesh3
